web/web.go

```diff
@@ -140,13 +140,13 @@ func newMetrics(r prometheus.Registerer) *metrics {
 	return m
 }
 
-func (m *metrics) instrumentHandlerWithPrefix(prefix string) func(handlerName string, handler http.HandlerFunc) http.HandlerFunc {
-	return func(handlerName string, handler http.HandlerFunc) http.HandlerFunc {
+func (m *metrics) instrumentHandlerWithPrefix(prefix string) func(handlerName string, handler http.Handler) http.HandlerFunc {
```

Comment:
Not a biggy, but it looks like this decision creates more issues than helping (same for `readyf`): adding `http.HandlerFunc(` everywhere. Why we need this? 🤔

Reply:
This makes the code follow the "accept interfaces, return concrete types" paradigm.

I think the benefit of this change will be seen in the future, as we break the api package down into multiple handlers. 

I was hesitant too, but @roidelapluie asked for it...  Now its in I like it.